python/lsst/obs/metadata/observationInfo.py

```diff
@@ -0,0 +1,190 @@
+                                                " (options: sky, unknown)", "str"),
+                   "detector_num": ("Unique (for instrument) integer identifier for the sensor", "int"),
+                   "detector_name": ("Name of the detector within the instrument (might not be unique)",
+                                     "str"),
+                   "detector_exposure_id": ("Unique integer identifier for this detector in this exposure",
+                                            "int"),
+                   "object": ("Object of interest or field name", "str"),
+                   "temperature": ("Temperature outside the dome", "astropy.units.Quantity"),
+                   "pressure": ("Atmospheric pressure outside the dome", "astropy.units.Quantity"),
+                   "relative_humidity": ("Relative humidity outside the dome", "float"),
+                   "tracking_radec": ("Requested RA/Dec to track", "astropy.coordinates.SkyCoord"),
+                   "altaz_begin": ("Telescope boresight azimuth and elevation at start of observation",
+                                   "astropy.coordinates.AltAz"),
+                   "science_program": ("Observing program (survey or proposal) identifier", "str"),
+                   "obstype": ("Type of observation (currently: science, dark, flat, bias, focus)",
+                               "str"),
+                   "obsid": ("Label uniquely identifying this observation (can be related to 'exposure')",
+                             "str")}
+
+    """All the properties supported by this class with associated
+    documentation."""
+
+    def __init__(self, header, translator_class=None):
+
+        # Store the supplied header for later stripping
+        self._header = header
+
+        # PropertyList is not dict-like so force to a dict here to simplify
+        # the translation code.
+        if hasattr(header, "toOrderedDict"):
+            header = header.toOrderedDict()
+
+        if translator_class is None:
+            translator_class = MetadataTranslator.determineTranslator(header)
+        elif not issubclass(translator_class, MetadataTranslator):
+            raise TypeError(f"Translator class must be a MetadataTranslator, not {translator_class}")
+
+        # Create an instance for this header
+        translator = translator_class(header)
+
+        # Store the translator
+        self._translator = translator
+
+        # Loop over each translation (not final form -- this should be
+        # defined in one place and consistent with translation classes)
+
+        for t in self._PROPERTIES:
+            # prototype code
+            method = f"to_{t}"
+            property = f"_{t}"
+
+            try:
+                setattr(self, property, getattr(translator, method)())
+            except AttributeError:
+                log.warning("No translation exists for property %s using translator %s",
+                            t, translator.__class__)
```

Comment:
I'm not sure if this should be `warning`, `error`, or `raise`. Warnings can be easy to miss.

Reply:
Raising might be better for this one because it's a triggered by the person implementing the translator rather than the person using the translation class. On the other hand it's annoying when implementing the class since it means that nothing runs until all translation methods have been implemented.

Reply:
I'm trying to think up with a work-around for that latter. On the other hand, "raise until everything works" *is* a way to help ensure that the final system is at least sensible (whether or not its correct).

Reply:
How about I provide a translator class that deliberately stubs them out with warnings and when you have finished your translator you remove that stub from the class hierarchy?